master/custom/factories.py

```diff
@@ -363,6 +363,34 @@ class NoBuiltinHashesUnixBuildExceptBlake2(UnixBuild):
     factory_tags = ["no-builtin-hashes-except-blake2"]
 
 
+class FedoraBuild(UnixBuild):
+    # Build on Fedora or RHEL.
+    #
+    # Try to be as close as possible to the Fedora specfile used to build
+    # the RPM package:
+    # https://src.fedoraproject.org/rpms/python3.10/blob/rawhide/f/python3.10.spec
+    configureFlags = [
+        "--with-pydebug",
+        "--with-platlibdir=lib64",
+        "--enable-ipv6",
+        "--enable-shared",
+        "--with-computed-gotos=yes",
+        "--with-dbmliborder=gdbm:ndbm:bdb",
+        "--with-system-expat",
+        "--with-system-ffi",
+        "--with-system-libmpdec",
```

Comment:
Adding options like this means we need to pull in the devel packages. Also, IIRC we can only use system mpdec in 3.10+

Reply:
The devel packages are already in the buildbots